### ISSUE TITLE
App.py is not working

### ISSUE THREAD

**Issue by anu0312** [Rogendo/Mental-health-Chatbot#4]
The error HFValidationError is raised because the provided checkpoint name ("./model/en_sw/") does not follow the expected format for Hugging Face model checkpoints.

When using the from_pretrained method, you should provide the name of a pre-trained model hosted on the Hugging Face Model Hub. If you have a custom model stored locally, you should use the from_pretrained method with the local_files_only argument set to True. @its-kios09 @Rogendo

**Comment by Rogendo** [comment]
Sorry you are encountering this error, there's a feature I was adding to the chatbot, I will provide a link where you can access the saved models. The retrained pre-trained model files are too large so I had to gitignore them.

**Comment by its-kios09** [comment]
@anu0312 There are new features that we adding on the bot

**Comment by anu0312** [comment]
> Sorry you are encountering this error, there's a feature I was adding to the chatbot, I will provide a link where you can access the saved models. The retrained pre-trained model files are too large so I had to gitignore them.

where is the link

**Comment by Rogendo** [comment]
We are still working on the feature, Even if we provide you with the files, there are some issues with the code. If you are in a rush, I can provide you with the app.py file that works but without the feature we are working on.

**Comment by Rogendo** [comment]
> We are still working on the feature, Even if we provide you with the files, there are some issues with the code. If you are in a rush, I can provide you with the app.py file that works but without the feature we are working on.

here is an older version of the file that works. This is the code

`import nltk
nltk.download('popular')#  Natural Language Toolkit (NLTK) library installed will download and
# install a collection of popular NLTK data packages, including corpora,
# models etc
from nltk.stem import WordNetLemmatizer
lemmatizer = WordNetLemmatizer()
import pickle
import numpy as np
from keras.models import load_model
model = load_model('model.h5')
import json
import random





intents = json.loads(open('intents.json').read())
words = pickle.load(open('texts.pkl','rb'))
classes = pickle.load(open('labels.pkl','rb'))
def clean_up_sentence(sentence):
    # tokenize the pattern - split words into array
    sentence_words = nltk.word_tokenize(sentence)
    # stem each word - create short form for word
    sentence_words = [lemmatizer.lemmatize(word.lower()) for word in sentence_words]
    return sentence_words
# return bag of words array: 0 or 1 for each word in the bag that exists in the sentence
def bow(sentence, words, show_details=True):
    # tokenize the pattern
    sentence_words = clean_up_sentence(sentence)
    # bag of words - matrix of N words, vocabulary matrix
    bag = [0]*len(words)  
    for s in sentence_words:
        for i,w in enumerate(words):
            if w == s: 
                # assign 1 if current word is in the vocabulary position
                bag[i] = 1
                if show_details:
                    print ("found in bag: %s" % w)
    return(np.array(bag))
def predict_class(sentence, model):
    # filter out predictions below a threshold
    p = bow(sentence, words,show_details=False)
    res = model.predict(np.array([p]))[0]
    ERROR_THRESHOLD = 0.25
    results = [[i,r] for i,r in enumerate(res) if r>ERROR_THRESHOLD]
    # sort by strength of probability
    results.sort(key=lambda x: x[1], reverse=True)
    return_list = []
    for r in results:
        return_list.append({"intent": classes[r[0]], "probability": str(r[1])})
    return return_list
def getResponse(ints, intents_json):
    tag = ints[0]['intent']
    list_of_intents = intents_json['intents']
    for i in list_of_intents:
        if(i['tag']== tag):
            result = random.choice(i['responses'])
                    
            break

    

    return result
def chatbot_response(msg):
    # Predict intent and get initial response in English
    ints = predict_class(msg, model)
    res = getResponse(ints, intents)
    # print(res)

   
    return res

    
# Creating GUI with flask
from flask import Flask, render_template, request
app = Flask(__name__)
app.static_folder = 'static'
@app.route("/")
def home():
    return render_template("index.html")

@app.route("/get")
def get_bot_response():
    userText = request.args.get('msg')
    print(userText)

    

    return chatbot_response(userText)
if __name__ == "__main__":
    app.run()`

**Comment by Rogendo** [comment]
> We are still working on the feature, Even if we provide you with the files, there are some issues with the code. If you are in a rush, I can provide you with the app.py file that works but without the feature we are working on.

Here is the code of an older version of app.py that works.
Delete all the contents of the app.py file and add the code below.





import nltk
nltk.download('popular')#  Natural Language Toolkit (NLTK) library installed will download and
# install a collection of popular NLTK data packages, including corpora,
# models etc
from nltk.stem import WordNetLemmatizer
lemmatizer = WordNetLemmatizer()
import pickle
import numpy as np
from keras.models import load_model
model = load_model('model.h5')
import json
import random


intents = json.loads(open('intents.json').read())
words = pickle.load(open('texts.pkl','rb'))
classes = pickle.load(open('labels.pkl','rb'))
def clean_up_sentence(sentence):
    # tokenize the pattern - split words into array
    sentence_words = nltk.word_tokenize(sentence)
    # stem each word - create short form for word
    sentence_words = [lemmatizer.lemmatize(word.lower()) for word in sentence_words]
    return sentence_words
# return bag of words array: 0 or 1 for each word in the bag that exists in the sentence
def bow(sentence, words, show_details=True):
    # tokenize the pattern
    sentence_words = clean_up_sentence(sentence)
    # bag of words - matrix of N words, vocabulary matrix
    bag = [0]*len(words)  
    for s in sentence_words:
        for i,w in enumerate(words):
            if w == s: 
                # assign 1 if current word is in the vocabulary position
                bag[i] = 1
                if show_details:
                    print ("found in bag: %s" % w)
    return(np.array(bag))
def predict_class(sentence, model):
    # filter out predictions below a threshold
    p = bow(sentence, words,show_details=False)
    res = model.predict(np.array([p]))[0]
    ERROR_THRESHOLD = 0.25
    results = [[i,r] for i,r in enumerate(res) if r>ERROR_THRESHOLD]
    # sort by strength of probability
    results.sort(key=lambda x: x[1], reverse=True)
    return_list = []
    for r in results:
        return_list.append({"intent": classes[r[0]], "probability": str(r[1])})
    return return_list
def getResponse(ints, intents_json):
    tag = ints[0]['intent']
    list_of_intents = intents_json['intents']
    for i in list_of_intents:
        if(i['tag']== tag):
            result = random.choice(i['responses'])
                    
            break

    

    return result
def chatbot_response(msg):
    # Predict intent and get initial response in English
    ints = predict_class(msg, model)
    res = getResponse(ints, intents)
    # print(res)

   
    return res

    
# Creating GUI with flask
from flask import Flask, render_template, request
app = Flask(__name__)
app.static_folder = 'static'
@app.route("/")
def home():
    return render_template("index.html")

@app.route("/get")
def get_bot_response():
    userText = request.args.get('msg')
    print(userText)

    

    return chatbot_response(userText)
if __name__ == "__main__":
    app.run()

**Comment by anu0312** [comment]
I cant host that local server even though there are no errors

**Comment by Rogendo** [comment]
> > Sorry you are encountering this error, there's a feature I was adding to the chatbot, I will provide a link where you can access the saved models. The retrained pre-trained model files are too large so I had to gitignore them.
> 
> where is the link

@anu0312 here is the link to access and download the models https://drive.google.com/drive/folders/1ybwgK1XNG1wd8As0m9vjMdQfHmD6E9uk?usp=sharing
